File: app/src/main/java/com/nhancv/sample/MainActivity.java
/*
 * Developed by Nhan Cao on 8/25/19 9:44 AM.
 * Last modified 8/25/19 9:44 AM.
 * Copyright (c) 2019. All rights reserved.
 */

package com.nhancv.sample;

import android.Manifest;
import android.os.Bundle;
import android.util.Log;

import com.nhancv.npermission.NDefaultPermission;
import com.nhancv.npermission.OnPermissionResult;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements OnPermissionResult {

    private NDefaultPermission nDefaultPermission;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // @nhancv 2019-08-25: request camera permission
        nDefaultPermission = new NDefaultPermission(MainActivity.this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        nDefaultPermission.forceAllPermissionGranted();
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        nDefaultPermission.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void onPermissionResult(@NotNull String permission, boolean isGranted) {
//        2019-08-25 21:54:12.240 16257-16257/com.nhancv.demo E/CAMERA: granted
//        2019-08-25 21:54:16.776 16257-16257/com.nhancv.demo E/WRITE_EXTERNAL_STORAGE: granted
//        2019-08-25 21:54:16.777 16257-16257/com.nhancv.demo E/READ_EXTERNAL_STORAGE: granted
    }
}

File: app/src/main/java/com/nhancv/sample/MainActivity.kt
/*
 * Developed by Nhan Cao on 8/25/19 9:44 AM.
 * Last modified 8/25/19 9:44 AM.
 * Copyright (c) 2019. All rights reserved.
 */

package com.nhancv.sample

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.nhancv.npermission.AllPermissionResult
import com.nhancv.npermission.NAllPermission


class MainActivity : AppCompatActivity(), AllPermissionResult {

    private lateinit var nAllPermission: NAllPermission

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // @nhancv 2019-08-25: Create permission controller
        nAllPermission = NAllPermission(this,
                allPermissionResult = this)
    }

    override fun onResume() {
        super.onResume()
        nAllPermission.forceAllPermissionGranted()
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        nAllPermission.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    override fun onAllPermissionGranted() {
        Log.e(TAG, "onAllPermissionGranted: isAllGranted")
    }

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }
}
